app/controllers/posts_controller.rb

```diff
@@ -4,8 +4,37 @@ def index
     @posts = Post.where(author_id: @user)
   end
 
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.author_id = current_user.id
+    @post.user_id = current_user.id
```

Comment:
- [ ] [OPTIONAL] Please remember that there is a has many relationships between the `User` and `Post` models. Therefore, you can take advantage of this like this 👇 
```ruby
@post = @current_user.posts.new(post_params)
```
Please do the same for the `CommentsController` and `LikesController`